sr_speech_control/src/sr_speech_control/speech_control.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+import time
+import speech_recognition as sr
+from difflib import get_close_matches
+from std_msgs.msg import String
+
+
+class SpeechControl(object):
+    def __init__(self, trigger_word, command_words, command_topic='speech_control',
+                 non_speaking_duration=0.1, pause_threshold=0.2):
+        self.trigger_word = trigger_word
+        self.command_words = command_words
+        self.recognizer = sr.Recognizer()
+        self.microphone = sr.Microphone()
+        self.command_publisher = rospy.Publisher(command_topic, String, queue_size=1)
+        self.command_to_be_executed = None
+
+        self._init_recognizer(non_speaking_duration, pause_threshold)
+        self._stop_listening = self.recognizer.listen_in_background(self.microphone, self._recognizer_callback)
+
+    def _init_recognizer(self, non_speaking_duration, pause_threshold):
+        with self.microphone as source:
+            self.recognizer.adjust_for_ambient_noise(source)
+        self.recognizer.non_speaking_duration = non_speaking_duration
+        self.recognizer.pause_threshold = pause_threshold
+
+    def _recognizer_callback(self, recognizer, audio):
+        try:
+            result = recognizer.recognize_google(audio)
+        except sr.UnknownValueError:
+            return
```

Comment:
Is this case when Google can't recognise audio? If yes, maybe worth adding comment as now is not clear why warning is not logged here but logged on other exception bellow.

Reply:
This is so the console doesn't get polluted by logs when user speaks random words or there is a background chatter. These commands should just be ignored, but adding information about them would not be a useful for the user in this case.